lib/streamlit/elements/widgets/chat.py

```diff
@@ -277,6 +278,13 @@ def chat_input(
         check_callback_rules(self.dg, on_submit)
         check_session_state_rules(default_value=default, key=key, writes_allowed=False)
 
+        id = compute_widget_id(
```

Comment:
A related question here: we have [a request](https://github.com/streamlit/streamlit/issues/7054) to allow changing the `placeholder` without resetting the widget itself. I thought that we might be able to allow this use-case in case the `key` is provided by the user, e.g.: 

```python
if key:
    id = compute_widget_id(
        "chat_input",
        user_key=key,
        max_chars=max_chars,
    )
else:
    id = compute_widget_id(
        "chat_input",
        user_key=None,
        placeholder=placeholder,
        max_chars=max_chars,
    )
```

Do you think this work, or are there any negative side-effects of doing this?

Reply:
Let's keep that to a separate PR. Doing it like that is fine on the technical side (custom components now does it this way), the question is if we should do it, which is a UX/product design matter.